src/MicrosoftTeams.ts

```diff
@@ -463,6 +501,43 @@ namespace microsoftTeams
             ensureInitialized(frameContexts.authentication);
 
             sendMessage("authentication.authenticate.failure", reason);
+            window.close();
+        }
```

Comment:
That was just a red herring from what I remember but I will add a timeout around the window.close() just in case.